apps/emqx_rule_engine/src/emqx_rule_actions.erl

```diff
@@ -110,8 +110,9 @@ republish(
     Payload = format_msg(PayloadTks, Selected),
     QoS = replace_simple_var(QoSTks, Selected, 0),
     Retain = replace_simple_var(RetainTks, Selected, false),
+    PubProps = maps:get(pub_props, OriginalMsg, #{}),
```

Comment:
This was originally (#9240) taken from `Selected`,
which seemed to be wrong because selected element keys are of binary format. (i.e. `<<"pub_props">>`).

Now if we take it from the original message, it may seem to work for republishes, however it also affects
other events like `message.dropped`, `message.acked`, in those events, a user may not necessary
want to keep the properties.

For example, if someone just want to republish a `message.dropped` event, but only interested in the topic name.

the "proper" approach is perhaps to treat message headers like qos, retain flag, and payload, 
meaning to add another argument here:
https://github.com/emqx/emqx/blob/966e6ddeeb9efb0d76c2f20fe3e7d1e44c4434a8/apps/emqx_rule_engine/src/emqx_rule_engine_schema.erl#L139-L177

@terry-xiaoyu @kjellwinblad @thalesmg wdyt ?

Reply:
Yes, we need a 'properties' config in this case. 

- If the 'properties' is not configured and it's a PUBLISH message, we republish the message using the original properties. 
- If the 'properties' is not configured and it's an event like `$events/client_connected`, we republish the message without any property. 
- If the 'properties' is configured, then we republish using the configured properties.

Reply:
From my very limited knowledge about this, it seems like the props should indeed be a new config; otherwise, it's not obvious that a property should be dropped or not.

Reply:
Thank you for your suggestions.
I have added a `user_properties` republish action argument with default template `${user_properties}`.

Decided *NOT* to fill `user_properties` with original message's `pub_props.'User-Property'`,
because the code is shared for both message republish and for message events such as `message.acked` and `message.dropped`.

In order to republish message with `User-Property`, a user will have to explicitly select a variable,
or configure with the special value `${pub_props.'User-Property'}`.